dust-tt/dust
# [connectors] Implement incremental sync for Zendesk tickets

## Description

- Close [#8549](https://github.com/dust-tt/dust/issues/8549)
- The `incremental` sync for Zendesk tickets refers to the use of the incremental endpoint, which retrieves a diff starting from a certain timestamp.
- This timestamp is stored and retrieved in a new table `ZendeskWorkspace`.
- As discussed in [#1546](https://github.com/dust-tt/tasks/issues/1546), in this PR this endpoint is queried on a cron schedule to keep data in sync.
- Documentation [here](https://developer.zendesk.com/documentation/ticketing/managing-tickets/using-the-incremental-export-api/#cursor-based-incremental-exports).
- Change the type of some columns in db from string to text to fix insertion.

- Tests performed:
  - Add a ticket In Zendesk and wait for 5 min (cron schedule frequency) to see if it appears in postgres + qdrant -> OK
  - Delete a ticket and see if it is deleted from postgres + qdrant -> OK

## Risk

Low

## Deploy Plan

- Deploy connectors.
